### PR TITLE
Ref task 3029 handle_with_item

### PR DESCRIPTION
**Pull request by manhphan34**
### Summary:
- [Feature] Profile  screen
### Link Issue:
- https://edu-redmine.framgia.vn/issues/3029

### CODE REVIEW

**Comment by manhphan34** [comment]
ready

File: app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java [hunk 1]
```diff
@@ -95,4 +249,30 @@ private void initActionBar() {
         getSupportActionBar().setDisplayShowHomeEnabled(true);
         getSupportActionBar().setDisplayShowTitleEnabled(false);
     }
+
+    private void initDialog() {
+        int width = (int) (getResources().getDisplayMetrics().widthPixels * 0.90);
+        int height = (int) (getResources().getDisplayMetrics().heightPixels * 0.70);
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java]: 254 255 hardcode

File: app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java [hunk 2]
```diff
@@ -95,4 +249,30 @@ private void initActionBar() {
...
+        mDialog.findViewById(R.id.button_choose_gallery).setOnClickListener(this);
+        mDialog.findViewById(R.id.button_update).setOnClickListener(this);
+    }
+
+    private String getText(EditText text) {
+        return text.getText().toString();
+    }
+
+    private void navigate() {
+        Intent intent = new Intent(this, LoginActivity.class);
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java]: convention 2.11

File: app/src/main/res/layout/dialog_edit_profile.xml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        android:text="@string/dialog_text_header_title"
+        android:textColor="@android:color/black"
+        android:textSize="@dimen/sp_20"
+        android:textStyle="bold"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="@+id/guide_line_top" />
+
+    <EditText
+        android:id="@+id/editText"
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/dialog_edit_profile.xml]: convention 3.2

File: app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java [hunk 4]
```diff
@@ -73,6 +94,130 @@ public void hideButton() {
...
+            case R.id.button_choose_gallery:
+                mPresenter.requestPermission(Manifest.permission.READ_EXTERNAL_STORAGE,
+                        PERMISSIONS_REQUEST_READ_EXTERNAL);
+                break;
+            case R.id.button_update:
+                mPresenter.updateInfo(getText(mEditDialogUserName));
+                break;
+            case R.id.button_sign_out:
+                signOut();
+                break;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java:218]: add default case

File: app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java [hunk 5]
```diff
@@ -73,6 +94,130 @@ public void hideButton() {
...
+    }
+
+    @Override
+    public void setImageDialogFromCamera(Bitmap bitmap) {
+        setImage(mDialogImage, bitmap);
+    }
+
+    @Override
+    public void chooseImageFromGallery() {
+        Intent pickPhoto = new Intent(Intent.ACTION_PICK,
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java]: convention 2.11

File: app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java [hunk 6]
```diff
@@ -1,27 +1,48 @@
...
+    public static final int PICK_IMAGE_FROM_GALLERY = 0x248;
+    public static final int PICK_IMAGE_FROM_CAMERA = 0x249;
+    public static final int PERMISSIONS_REQUEST_READ_EXTERNAL = 0x141;
+    public static final int PERMISSIONS_REQUEST_CAMERA = 0x142;
     private static final String EXTRA_USER = "EXTRA_USER";
     private static final String SPACE = " ";
     private TextView mTextUserName, mTextEmail, mTextLastSignIn;
     private ImageView mImageCover, mImageViewAvatar;
     private Button mButtonEdit, mButtonSignOut;
     private ProfileContract.Presenter mPresenter;
```
Comment (NguyenMinhTuanB) [inline app/src/main/java/framgia/com/ichat/screen/profile/ProfileActivity.java:51]: 39 -> 42 mỗi biến nên để 1 dòng

**Comment by NguyenMinhTuanB** [comment]
commented

**Comment by manhphan34** [comment]
ready

File: app/src/main/res/layout/dialog_edit_profile.xml [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        android:id="@+id/guide_line_top"
+        android:layout_width="@dimen/dp_0"
+        android:layout_height="@dimen/dp_0"
+        android:orientation="horizontal"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintGuide_percent="0.05"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+
```
Comment (NguyenMinhTuanB) [inline app/src/main/res/layout/dialog_edit_profile.xml]: remove bl

**Comment by manhphan34** [comment]
ready

**Comment by NguyenMinhTuanB** [comment]
conflicted

**Comment by manhphan34** [comment]
ready

**Pull request merged**